embedding.py

```diff
@@ -0,0 +1,54 @@
+import csv
+from sentence_transformers import SentenceTransformer
+import torch
+
+
+def combine_2_embeddings(embd_type, text, title):
+    if embd_type == "concatenate":
+        x = torch.cat((text, title))
+    elif embd_type == "subtract":
+        x = torch.sub(text, title)
+    else:
+        raise ValueError(f"Invalid embedding type: {embd_type}. Should be concatenate or subtract")
+    return x
+
+
+def get_incongruent_paragraphs(label, incongruent_sentences):
+    incongruent_paragraphs = []
+    if label == 1:
+        if len(incongruent_sentences) > 2:
+            incongruent_sentences = incongruent_sentences[1:-1].split(', ')
+            for el in incongruent_sentences:
+                incongruent_paragraphs.append(int(el))
+    return incongruent_paragraphs
+
+
+def get_x_y(**embedding):
+    file, tokenization_type, embeddings, model_name = [embedding[el] for el in embedding]
+    model = SentenceTransformer(model_name)
+    X_train = []
+    y_train = []
+    pattern = "<EOP>"
+    with open(file, encoding='utf-8') as f:
+        csvreader = csv.reader(f, delimiter='\t')
+        for j, (_, title, body, label, _, incongruent_sentences, _) in enumerate(csvreader):
+            title_embedding = model.encode(title, convert_to_tensor=True)
+            body = body.strip().split(pattern)
+            incongruent_paragraphs = get_incongruent_paragraphs(int(label), incongruent_sentences)
+            for i, element in enumerate(body):
+                x = []
+                y = []
+                if tokenization_type == "<EOS>":
+                    sentences = element.split("<EOS>")
+                    for sentence in sentences:
+                        sentence_embedding = model.encode(sentence, convert_to_tensor=True)
+                        x.append(combine_2_embeddings(embeddings, sentence_embedding, title_embedding))
+                        y.append(int(i in incongruent_paragraphs))
+
+                else:
+                    body_embedding = model.encode(element, convert_to_tensor=True)
+                    x.append(combine_2_embeddings(embeddings, body_embedding, title_embedding))
+                    y.append(int(i in incongruent_paragraphs))
```

Comment:
Lines 44-46 and 49-51 contain code duplication. 

```suggestion
                if tokenization_type == "<EOS>":
                    sentences = element.split("<EOS>")
                else:
                    sentences = element
                    
                for sentence in sentences:
                     sentence_embedding = model.encode(sentence, convert_to_tensor=True)
                     x.append(combine_2_embeddings(embeddings, sentence_embedding, title_embedding))
                     y.append(int(i in incongruent_paragraphs))
```